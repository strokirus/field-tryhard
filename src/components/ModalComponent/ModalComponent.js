import React from 'react';
import { object, bool, func, string } from 'prop-types';
import Modal from 'react-responsive-modal';

const ModalComponent = ({ open, onCloseModal, title, info }) => (
  <Modal
    open={open}
    onClose={onCloseModal}
  >
    <h2
      className="title-modal"
    >
      {title}
    </h2>

    {Object.keys(info).map((s, i) => {
      if (s !== 'name' && typeof info[s] !== 'object') {
        return (<p key={`info-${i}-${s}`}>{`${s}: ${info[s]}`}</p>);
      } else if (typeof info[s] === 'object') {
        return (Object.keys(info[s]).map(e => (
          <p key={`info-${i}-${s}`}>{`${e}: ${info[s][e]}`}</p>
        )));
      }

      return ('');
    })}
  </Modal>
);

ModalComponent.propTypes = {
  open: bool,
  onCloseModal: func,
  title: string,
  info: object,
};

ModalComponent.defaultProps = {
  open: false,
  onCloseModal: () => {},
  title: '',
  info: { },
};

export default ModalComponent;
