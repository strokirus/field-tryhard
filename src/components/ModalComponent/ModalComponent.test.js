import React from 'react';
import renderer from 'react-test-renderer';
import ModalComponent from './ModalComponent';

const props = { };

describe('(Component) ModalComponent', () => {
  it('should render an ModalComponent with result', () => {
    const tree = renderer.create(<ModalComponent {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
