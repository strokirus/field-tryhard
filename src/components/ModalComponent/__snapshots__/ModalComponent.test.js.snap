// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ModalComponent should render an ModalComponent with result 1`] = `null`;
